tools/src/main/java/org/apache/kafka/trogdor/workload/SustainedConnectionWorker.java

```diff
@@ -0,0 +1,563 @@
+        this.doneFuture = doneFuture;
+        this.status = status;
+        this.connections = new ArrayList<>();
+
+        // Initialize all status reporting metrics to 0.
+        this.totalProducerConnections = new AtomicLong(0);
+        this.totalProducerFailedConnections = new AtomicLong(0);
+        this.totalConsumerConnections = new AtomicLong(0);
+        this.totalConsumerFailedConnections = new AtomicLong(0);
+        this.totalMetadataConnections = new AtomicLong(0);
+        this.totalMetadataFailedConnections = new AtomicLong(0);
+        this.totalAborts = new AtomicLong(0);
+
+        // Create the worker classes and add them to the list of items to act on.
+        for (int i = 0; i < this.spec.producerConnectionCount(); i++) {
+            this.connections.add(new ProducerSustainedConnection());
+        }
+        for (int i = 0; i < this.spec.consumerConnectionCount(); i++) {
+            this.connections.add(new ConsumerSustainedConnection());
+        }
+        for (int i = 0; i < this.spec.metadataConnectionCount(); i++) {
+            this.connections.add(new MetadataSustainedConnection());
+        }
+
+        // Create the status reporter thread and schedule it.
+        this.statusUpdaterExecutor = Executors.newScheduledThreadPool(1,
+            ThreadUtils.createThreadFactory("StatusUpdaterWorkerThread%d", false));
+        this.statusUpdaterFuture = this.statusUpdaterExecutor.scheduleAtFixedRate(
+            new StatusUpdater(), 0, REPORT_INTERVAL_MS, TimeUnit.MILLISECONDS);
+
+        // Create the maintainer pool, add all the maintainer threads, then start it.
+        this.workerExecutor = Executors.newFixedThreadPool(spec.numThreads(),
+            ThreadUtils.createThreadFactory("SustainedConnectionWorkerThread%d", false));
+        for (int i = 0; i < this.spec.numThreads(); i++) {
+            this.workerExecutor.submit(new MaintainLoop());
+        }
+    }
+
+    private interface SustainedConnection extends AutoCloseable {
+        boolean needsRefresh();
+        void refresh();
+        void claim();
+    }
+
+    private abstract class ClaimableConnection implements SustainedConnection {
+
+        // These variables are used to maintain the scheduling of the connection keep-alive.
+        protected long nextUpdate = 0;
+        protected boolean inUse = false;
+
+        @Override
+        public boolean needsRefresh() {
+            return !this.inUse && (System.currentTimeMillis() > this.nextUpdate);
+        }
+
+        @Override
+        public void claim() {
+            this.inUse = true;
+        }
+
+        @Override
+        public void close() throws Exception {
+            this.closeQuietly();
+        }
+
+        protected abstract void closeQuietly();
+
+    }
+
+    private class MetadataSustainedConnection extends ClaimableConnection {
+
+        // This variable is used to maintain the connection itself.
```

Comment:
Some of these comments seem superfluous. I think it would be better if we put a javadoc right on top of `private class MetadataSustainedConnection` to say something like

"This class maintains connections through an AdminClient. It does so by calling the DescribeCluster Admin API at an interval of refreshRateMs."
That would explain most of the variables in a more succinct way

Reply:
Yeah, they made much more sense when I was coding it because I had a lot more, but then I consolidated and it just became noise.

Reply:
Check out what I've done in my next commit (when I post it) and let me know what you think.